RegExp/regExp.js

```diff
@@ -0,0 +1,65 @@
+const localeReg = "bg_BG".match(/[a-z]{2}_[A-Z]{2}?/);
+console.log(localeReg);
+
+const passReg = "fd a$sFg%G34 ghg".match(/[\w\s*&$#+=%^@]{8,64}/);
+const passBonusReg = "fd a$sFg%35Ghg".match(/(?=.*[a-z])(?=.*\d)(?=.*[ _*&$#+=%^@)(])?(?=.*[A-Z]).{8,64}/);
+console.log(passBonusReg);
+
+const phoneReg = "+35982502010".match(/^([+]359|0)\d{6,9}/);
+console.log(phoneReg);
+
+const emailReg = "my_unique_mail.01+comment@sub.domain.example.com".match(/([\w.+]+)@([a-z.]+)(.[a-zA-Z]{2,}$)/);
+console.log(emailReg);
+
+const nameReg = "Ана-Мария Димитрова Петкова";
+const regex = /([а-яА-Я-.]+\s)([а-яА-Я-.]+\s)?([а-яА-Я-.]+)/;
+console.log(nameReg.match(regex));
+const newName = nameReg.replace(regex, "$3, $1");
+console.log(newName);
+
+const urlReg = "ftp://johnsmith@my.domain.com:8080/path/to/resource/file.txt";
+const regexUrl = /^([a-z]+):[/]{2}([w]{3})?\.?([a-z.]+@)?\/?([a-z.]+):?(\d+)?\/?([/\w]+[/|?])?([\w]+[.a-z]+)?([?\w=&]+)?/;
```

Comment:
Can you extract the : host, username, domain, port, path, filename, extension, query?